packages/oui-criteria-container/src/criteria-container.controller.js

```diff
@@ -39,6 +39,16 @@ export default class CriteriaController {
     }
 
     add (criterion) {
+        // Delete same preview criterion if it exists.
+        const previewCriterion = cloneDeep(criterion);
```

Comment:
at a quick glance, I find that there is a lot of preventive/defensive copying using cloneDeep. 
eventually this could get expensive performance wise. perhaps we should consider immutablejs.

Reply:
In fact I think cloneDeep isn't necessary... It should work with a simple clone.
The performance is not really impacted because we'll never have a filter with a huge number of criteria.
ImmutableJS could be interesting for more specific usage but also at a cost of a significant increasing of dist size unfortunately.